docs/DesignDocs/Enabledebugmalloc.md

```diff
@@ -0,0 +1,154 @@
+Proposal to enable debug malloc for end users
+===
+
+# Motivation
+
+Debug malloc is an internal tool which can help identify memory leaks in enclaves.
+It is turned on by default in debug builds of the SDK, and helps detect leaks in the SDK.
+
+the main logic of Debug malloc is as following:
+- Whenever memory is allocated, it captures the callstack and associates it with the allocated memory.
+  Additionally it also sets the contents of the allocated memory to zero.
+
+- Whenever memory is freed, it clears the contents to a fixed pattern to help detect access of freed memory.
+  It also frees the associated callstack information.
+
+- When enclave terminates, it prints information about leaks - those objects that have been allocated,
+  but not feed.
+
+Debug malloc is performance intensive and is therefore not compiled into the SDK release.
+
+There have been requests from users that debug malloc be made available so that they can
+identify leaks within the enclave. Commonly used memory detection tools like Valgrind do not
+work within the enclaves since they cannot access enclave memory.
```

Comment:
Have we looked at enabling LeakSanitizer/AddressSanitizer in enclave?

Reply:
Thanks, LeakSanitizer may be a good option. Now I just follow the fulfilled debug malloc and not much changes involved.